frontend/src/components/PsyListing/PsyListing.jsx

```diff
@@ -1,115 +1,77 @@
 import React, { useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
 
-import Ariane from 'components/Ariane/Ariane';
 import agent from 'services/agent';
+import Page from 'components/Page/Page';
+import { Checkbox, TextInput } from '@dataesr/react-dsfr';
 import PsyTable from './PsyTable';
 
 const PsyListing = () => {
+  const query = new URLSearchParams(useLocation().search);
+
   const [psychologists, setPsychologists] = useState([]);
-  const [filter, setFilter] = useState({});
+  const [filter, setFilter] = useState(query.get('search') || '');
+  const [teleconsultation, setTeleconsultation] = useState(query.get('teleconsultation') || false);
 
   useEffect(() => {
     agent.Psychologist.find().then(response => {
       setPsychologists(response.psyList);
     });
   }, []);
 
-  const filterByKey = (psychologist, key, psyKey) => {
-    const filterValue = filter[key];
-    if (!filterValue) {
-      return true;
+  const getFilteredPsychologists = () => {
+    let filteredPsychologists = psychologists;
+    if (teleconsultation) {
+      filteredPsychologists = filteredPsychologists.filter(psychologist => psychologist.teleconsultation);
     }
-    const psychologistValue = psychologist[psyKey || key];
-    return psychologistValue && psychologistValue.toLowerCase().includes(filterValue);
-  };
 
-  const getFilteredPsychologists = () => psychologists.filter(
-    psychologist => filterByKey(psychologist, 'name', 'lastName')
-        && filterByKey(psychologist, 'address')
-        && filterByKey(psychologist, 'postCode', 'departement'),
-  );
+    if (!filter) {
+      return filteredPsychologists;
+    }
 
-  const changeFilter = (name, event) => {
-    const newFilter = { ...filter };
-    newFilter[name] = event.target.value.toLowerCase();
-    setFilter(newFilter);
+    const psychologistByName = filteredPsychologists
+      .filter(
+        psychologist => psychologist.lastName.toLowerCase().includes(filter.toLowerCase()),
+      );
+    const psychologistByAddress = filteredPsychologists.filter(
+      psychologist => (
+        !psychologist.lastName.toLowerCase().includes(filter.toLowerCase())
+        && (
+          psychologist.address.toLowerCase().includes(filter.toLowerCase())
+        || psychologist.departement.toLowerCase().includes(filter.toLowerCase())
+        || psychologist.region.toLowerCase().includes(filter.toLowerCase())
```

Comment:
`.toLowerCase().includes(filter.toLowerCase())` appears multiple time here and could have been extracted in a dedicated function